keps/sig-node/5328-node-capabilities/README.md

```diff
@@ -0,0 +1,864 @@
+
+###### Does enabling the feature change any default behavior?
+
+Yes. Introducing new capabilities can prevent pod scheduling or pod request admission when the nodes cannot satisfy the pod requirements. 
+*Note:* Enabling the NodeCapabilities feature gate alone doesn't alter default behavior; it only activates the underlying framework. Actual changes to behaviors like pod scheduling or admission happen once nodes start publishing specific capabilities (for instance, Kubelet reporting a feature gate's status as a capability) and control plane components are updated to consume and act on this information.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled, but a specific rollback order must be followed to prevent scheduling disruptions. The feature must be disabled on control plane components before being Disabled on nodes. When the feature is disabled on a Node, the Kubelet would clear the NodeCapabilities field from the Node's status. Existing pods that are already running are unaffected by this process.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+If the feature is re-enabled after being previously disabled, any new pod scheduling or request validation will again start considering Node Capabilities. The feature should be enabled on the nodes before it is enabled on the control plane to ensure the scheduling decisions are not affected.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+Yes, feature enablement/disablement tests will be added along with alpha implementation.
+
+* Validate that the kube-scheduler plugin filters nodes based on `node.status.capabilities` when the feature is enabled, and ignores the field when the feature is disabled.
+* Validate that `node.status.capabilities` is correctly populated when the feature is enabled, and the field is cleared from the `Node` object when the feature is disabled.
+* Validate that the Admission Controller correctly fetches and validates requests against capabilities when the feature is enabled, and does not block requests if the feature is disabled.
```

Comment:
Those are all good tests, but these are feature tests - not enablement/disablement.

Enablement/disablement is a test that (as stated in the comment in the template above) that switches the feature gate in the middle of the test.